airflow/providers/ftp/hooks/ftp.py

```diff
@@ -58,7 +58,11 @@ def get_conn(self) -> ftplib.FTP:
         if self.conn is None:
             params = self.get_connection(self.ftp_conn_id)
             pasv = params.extra_dejson.get("passive", True)
-            self.conn = ftplib.FTP(params.host, params.login, params.password)  # nosec: B321
+            self.conn = ftplib.FTP()  # nosec: B321
+            if params.host:
+                self.conn.connect(params.host, params.port)
+                if params.login:
+                    self.conn.login(params.login, params.password)
```

Comment:
Can `params.login` be provided but not `params.host`? If yes, then there is an indentention issue

Reply:
This is implemented in the same way as in the standard library. Plus, `params.login` and `params.password` only have a meaning when trying to connect to a FTP server for which you need to have a host name supplied (even if it's localhost).

Reply:
No problem then